R/aggregate_results.R

```diff
@@ -16,63 +16,121 @@
 #'   arguments.
 #' @param iter_var String vector holding the variable over which to iterate in
 #'   sensitivity analysis
+#' @param risk_type String that is either lrisk or trisk
 #'
 #' @return A list including basic and aggregated results.
-aggregate_results <- function(results_list, sensitivity_analysis_vars, iter_var) {
+aggregate_results <- function(results_list, sensitivity_analysis_vars, iter_var, risk_type) {
   sensitivity_analysis_vars <- paste0(sensitivity_analysis_vars, "_arg")
 
   # Aggregate Crispy Results -----
 
-  crispy_output <- results_list$company_technology_npv %>%
-    dplyr::inner_join(
-      results_list$company_pd_changes_overall,
-      by = c(
-        "investor_name", "portfolio_name", "scenario_name", "scenario_geography",
-        "company_name", "ald_sector", "asset_type_arg", "baseline_scenario_arg",
-        "shock_scenario_arg", "lgd_arg", "risk_free_rate_arg", "discount_rate_arg",
-        "growth_rate_arg", "div_netprofit_prop_coef_arg", "shock_year_arg",
-        "fallback_term_arg", "use_company_terms_arg"
+  if (risk_type == "trisk") {
```

Comment:
Your code is all correct but induces a lot of duplicates code, which makes it a bit harder to read/to maintain. 

To improve this I would suggest that for all behaviors that depend on risk type you only put the aspect that differs into an if-else logic, or even just an if logic.

E.g. here you could:
1. create a vector merge_by_cols <- ("investor_name", "portfolio_name",...) for trisk
2. append the new variables to that vector if risk_type is "lrisk"
3.  a pass this vector to by in the inner_join. 

Similar simplifications could be made also for the other if else logics 
(tipp: You will sometimes have to break the pipes to simplify the code)

Reply:
hope this is addressed